src/main/java/seedu/address/logic/commands/AssignCommand.java

```diff
@@ -67,12 +70,18 @@ public CommandResult execute(Model model) throws CommandException {
             throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
         }
 
-        Driver driver = model.getDriver(driverId);
         Task task = model.getTask(taskId);
+        if (task.getStatus() != TaskStatus.INCOMPLETE || task.getDriver().isPresent()
+                || task.getEventTime().isPresent()) {
+            throw new CommandException(MESSAGE_ALREADY_ASSIGNED);
+        }
```

Comment:
having the task.getDriver().isPresent() || task.getEventTime().isPresent() seems like saying there's a possibility that only 1 of the field is populated which shouldn't be possible. I suggest creating a boolean method in task to check both at the same time.

Reply:
```
task.getStatus() != TaskStatus.INCOMPLETE && task.getDriver().isPresent() && task.getEventTime().isPresent()
```
Should be this right?

Reply:
It's more like defensive programming. If a task is not ongoing, then the driver and eventTime should be absent, only if the other classes are implemented correctly. Here I'm failing the command if any of those predicate is true.